proposals/lotus-blockstore-interface.md

```diff
@@ -0,0 +1,44 @@
+# Extended Blockstore Interface
+
+Authors: vyzo
+
+## What is the problem this project solves?
+
+We currently have a limited "dumb" blockstore interface, which does not allow us to pass context options so as to inform the blockstore about certain properties of objects being stored.
+
+This is particularly important in the context of the splitstore, where we have the following problems:
+- We lack the ability to inform the splitstore of the actual epoch an object belongs, which requires us to "guess" by estimating based on the current tipset and wall clock (see [#6474](https://github.com/filecoin-project/lotus/pull/6474)). This is problematic for many reasons, as an object might be misassigned its epoch (and thus become completely broken during catch-up sync in current master).
```

Comment:
This is very much the wrong way to go about this. Instead, the blockstore should be _managed_ by a system on-top of the blockstore that understands what we need to keep and what we don't need to keep.

It sounds like we need some form of tagged and/or generational blockstore:

1. Low-level blockstore that stores only blocks.
2. Intermediate service with a `ForEpoch(epoch) Blockstore` function that returns a blockstore for _that_ epoch.